src/aics-image-viewer/components/App/index.tsx

```diff
@@ -293,20 +295,14 @@ const App: React.FC<AppProps> = (props) => {
 
   // Image loading/initialization functions ///////////////////////////////////
 
-  const onChannelDataLoaded = (
-    aimg: Volume,
-    thisChannelsSettings: ChannelState,
-    channelIndex: number,
-    keepLuts = false
-  ): void => {
-    // if we want to keep the current control points
-    if (thisChannelsSettings.controlPoints && keepLuts) {
-      const lut = controlPointsToLut(thisChannelsSettings.controlPoints);
-      aimg.setLut(channelIndex, lut);
-    } else {
-      // need to choose initial LUT
+  const onChannelDataLoaded = (aimg: Volume, thisChannelsSettings: ChannelState, channelIndex: number): void => {
+    // If this is the first load of this image, auto-generate initial LUTs
+    if (initialLoadRef.current || !thisChannelsSettings.controlPoints) {
       const newControlPoints = initializeLut(aimg, channelIndex, props.viewerChannelSettings);
       changeChannelSetting(channelIndex, "controlPoints", newControlPoints);
+    } else {
+      const lut = controlPointsToLut(thisChannelsSettings.controlPoints);
+      aimg.setLut(channelIndex, lut);
```

Comment:
i guess the old `keepLuts` was unused?

Reply:
The value which was passed in as `keepLuts` was set outside the closure in which it was captured and used, so it always had a stale value (which was always `false`).